src/components/DataViewFilterText/DataViewFilterText.tsx

```diff
@@ -119,8 +119,13 @@ function DataViewFilterText(props: DataViewFilterTextProps) {
 	// 	throwOnInvalid : true
 	// });
 
+	if (props.comparisonDefault && validComparisons.find( validComparison => validComparison.value === props.comparisonDefault) === undefined)
+		throw new Error("The selected comparison is not a valid comparison")
+
 	const [anchorEl, setAnchorEl] = useState(null);
-	const comparison = props.data.comparison || (props.comparisonDefault || "equals");
+	const comparison = props.args && props.args.comparisons

```

Comment:
I think this is confusing and can be cleaned a lot. I'm not sure what the difference is between props.data.comparison || (props.comparisonDefault || "equals") : props.data.comparison || "equals"; it seems that can be rethinked.